j5/boards/base.py

```diff
@@ -104,6 +104,12 @@ def make_safe(self) -> None:
         for board in self.boards.values():
             board.make_safe()
 
+    def __str__(self) -> str:
+        """A string representation of the board group."""
+        list_str = ' , '.join(map(str, self.boards.values()))
```

Comment:
I feel like the space before the comma here will look weird, but that's just my preference.
```suggestion
        list_str = ', '.join(map(str, self.boards.values()))
```